tfjs-converter/python/tensorflowjs/converters/keras_h5_conversion.py

```diff
@@ -255,6 +389,49 @@ def h5_weights_to_tfjs_format(h5file, split_by_layer=False):
         groups[0] += group
   return groups
 
+def h5_v3_weights_to_tfjs_format(h5file, metafile, split_by_layer=False):
+  """Load weight values from a Keras HDF5 file and to a binary format.
+
+  The HDF5 file is one generated by Keras' Model.save_weights() method.
+
+  N.B.:
+  1) This function works only on HDF5 values from Keras version 2.
+  2) This function does not perform conversion for special weights including
+      ConvLSTM2D and CuDNNLSTM.
+
+  Args:
+    h5file: An instance of h5py.File, or the path to an h5py file.
+    metafile: A path of metadata.json file.
+    split_by_layer: (Optional) whether the weights of different layers are
+      to be stored in separate weight groups (Default: `False`).
+
+  Returns:
+    An array of group_weights as defined in tfjs write_weights.
+
+  Raises:
+    ValueError: If the Keras version of the HDF5 file is not supported
+  """
+  h5file = _ensure_h5file(h5file)
+  try:
+    _check_v3_version(metafile)
+  except ValueError:
+    print("""failed to lookup keras version from the meta file.""")
+
+  groups = [] if split_by_layer else [[]]
+
+  # pylint: disable=not-an-iterable
+  layer_names = [as_text(n) for n in h5file.attrs['layer_names']]
+  # pylint: enable=not-an-iterable
+  for layer_name in layer_names:
+    layer = h5file[layer_name]
+    group = _convert_h5_group(layer)
+    if group:
+      if split_by_layer:
+        groups.append(group)
+      else:
+        groups[0] += group
+  return groups
```

Comment:
This looks very similar to the weights extraction in `h5_v3_merged_saved_model_to_tfjs_format`. Would it make sense to call this function from there, or do the keys of the h5file not match up?

Reply:
Removed since not needed